Form/Extension/BooleanExtension.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Form\Extension;
+
+use FOS\RestBundle\Form\Transformer\BooleanTypeToBooleanTransformer;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class BooleanExtension extends AbstractTypeExtension
+{
+    const TYPE_API = 'api';
+    const TYPE_CHECKBOX = 'checkbox';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['type'] === self::TYPE_API) {
```

Comment:
I agree. The option could then also be changed to only except boolean values.

What do you also think about adding a global option to change the default value for this option? For example, in an API only application you might want to always set the option to `true` without changing all your form types while you rather do not want to do the same in an application that offers an API and a common HTML interface.